File: app/src/main/java/com/ragdroid/dahaka/activity/items/ItemsComponent.java
package com.ragdroid.dahaka.activity.items;



import com.ragdroid.dahaka.activity.ActivityScope;

import dagger.BindsInstance;
import dagger.Subcomponent;

/**
 * Created by garimajain on 17/08/17.
 */
@Subcomponent(modules = ItemsModule.class)
public interface ItemsComponent {

    void inject(ItemsActivity itemsActivity);

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        ItemsComponent.Builder activity(ItemsActivity homeActivity);
        ItemsComponent build();

    }
}

File: app/src/main/java/com/ragdroid/dahaka/activity/items/ItemsComponent.kt
package com.ragdroid.dahaka.activity.items


import dagger.BindsInstance
import dagger.Subcomponent

/**
 * Created by garimajain on 17/08/17.
 */
@Subcomponent(modules = arrayOf(ItemsModule::class))
interface ItemsComponent {

    fun inject(itemsActivity: ItemsActivity)

    @Subcomponent.Builder
    interface Builder {

        @BindsInstance
        fun activity(homeActivity: ItemsActivity): ItemsComponent.Builder

        fun build(): ItemsComponent

    }
}
